docs/usage.rst

```diff
@@ -333,25 +333,73 @@ configuration.
 Debugging Python functions
 --------------------------
 
-Unlike Node.JS and Java, Python requires you to enable remote debugging
-in your Lambda function code. If you enable debugging with
-``--debug-port`` or ``-d`` for a function that uses one of the Python
-runtimes, SAM CLI will just map through that port from your host machine
-through to the Lambda runtime container. You will need to enable remote
-debugging in your function code. To do this, use a python package such
-as `remote-pdb <https://pypi.python.org/pypi/remote-pdb>`__. When
-configuring the host the debugger listens on in your code, make sure to
-use ``0.0.0.0`` not ``127.0.0.1`` to allow Docker to map through the
-port to your host machine.
-
-   Please note, due to a `open
-   bug <https://github.com/Microsoft/vscode-python/issues/71>`__ with
-   Visual Studio Code, you may get a
-   ``Debug adapter process has terminated unexpectedly`` error when
-   attempting to debug Python applications with this IDE. Please track
-   the `GitHub
-   issue <https://github.com/Microsoft/vscode-python/issues/71>`__ for
-   updates.
+Python debugging requires you to enable remote debugging in your Lambda function code, therefore it's a 2-step process:
+
+1. Install ptvsd library and enable within your code
+2. Configure your IDE to connect to the debugger you configured for your function
+  - SAM CLI debug port option ``--debug-port`` or `-d` will map that port to the local Lambda container execution your IDE needs to connect to
+
+Ptvsd configuration
+^^^^^^^^^^^^^^^^^^^
+
+Provided that your source code has ptvsd available for your Lambda function you can do the following to enable the debugger:
+
+.. code:: python
+
+  import ptvsd
+  ...
+
+  def lambda_handler(event, context):
+    ...
+    ptvsd.enable_attach(address=('0.0.0.0', 5890), redirect_output=True)
+    ptvsd.wait_for_attach()
+
+**0.0.0.0** instead of **localhost** for listening across all network interfaces and **5890** is the debugging port of your preference.
+
+Visual Studio Code
+^^^^^^^^^^^^^^^^^^
+
+In order to setup Visual Studio Code (VS Code) for debugging with AWS SAM CLI, use
+the following launch configuration:
+
+.. code:: json
+
+   {
+       "version": "0.2.0",
+       "configurations": [
+           {
+              "name": "SAM CLI Python Hello World",
+              "type": "python",
+              "request": "attach",
+              "port": 5890,
+              "host": "localhost",
+              "pathMappings": [
+                  {
+                      "localRoot": "${workspaceFolder}/hello_world/build",

```

Comment:
You state to use this launch config to get debugging working but this localRoot is specific to our init example. Without understanding how to set this, customers that are not using the hello world init example may have difficulty.

My suggestion is to make explicit steps to help customers. Something along the lines:

There are X steps to get Python Debugging working with VSCode.

1. Install ptvsd into your project. This should be part of your dependencies that are mounted into the container.
`pip install ptvsd -t <path to project>` You can also add it as a 'dev dependency' and install them into the project folder like you would normally.
2. Open your app in vscode.
3. Copy 'this mostly complete' launch config to your launch config for your project.
4. Update 'configurations.pathMappings.localRoot' value to be ${workspaceFolder}/<path to the function code>
Note: the <path to the function code> should be the path to the folder that will be mounted to the docker container. If you have opened vscode in your project that has your `template.yml` then <path to the function code> should match your `CodeUri` on your function
5. Run sam local:
6. Launch the debugger

I think this gives a better way to understand the steps and make them very reproducible.

If you think the current structure is better, I ask to make it clear what needs to change and how to change it to get your function debuggable instead of saying this config will just make it work.

Reply:
Fair point. Reworded as a walkthrough style and used `sam init --runtime python3.6..` as a starting point and provided explicit step-by-step from there.

Reply:
* Pushed extended wording explaining step by step
* Moved generic wording under Debugging functions section as they're the same for other runtimes (explanation about `-d, --debug-port`). 

This line and onwards expand on the additional ask: https://github.com/awslabs/aws-sam-cli/pull/694/files#diff-6b8768d928ef9ab4f13707da19516369R339